app/presenters/project_show_presenter.rb

```diff
@@ -148,7 +148,7 @@ def quota_percentage(session_id:)
 
   def user_has_access?(user:)
     return true if user.eligible_sysadmin?
-    data_sponsor.uid == user.uid || data_manager.uid == user.uid || data_users.map(&:uid).include?(user.uid)
+    data_sponsor&.uid == user.uid || data_manager&.uid == user.uid || data_users.map(&:uid).include?(user.uid)
```

Comment:
This is really the fix: if the user does not have access to a project we cannot fetch the `data_sponsor` and we were getting into a `nil` exception. Now the code will return false which will prevent the access to the project.